src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -250,6 +250,13 @@ Copyright (c) .NET Foundation. All rights reserved.
       <RuntimePackAsset Remove="@(_ConflictPackageFiles)" Condition="'%(_ConflictPackageFiles.ConflictItemType)' != 'Reference'" />
     </ItemGroup>
 
+    <!-- Set valid RID platforms for runtime assets if targeting .NET 8+ and using the RID graph is not enabled. -->
+    <ItemGroup Condition="'$(_TargetFrameworkVersionWithoutV)' &gt;= '8.0'
+        and @(RuntimeHostConfigurationOption->WithMetadataValue('Identity', 'System.Runtime.Loader.UseRidGraph')->WithMetadataValue('Value', 'true')->Count()) == 0">
+        <_ValidRuntimeIdentifierPlatformsForAssets Include="any;freebsd;illumos;linux;linux-bionic;linux-musl;osx;solaris;unix;win" />
+        <_ValidRuntimeIdentifierPlatformsForAssets Include="android;aot;browser;ios;iossimulator;maccatalyst;tvos;tvossimulator" />
```

Comment:
How will this work in source build for some custom distro. In my mental model if I create a `amazetux` distro and setup source build for it, the SDK should allow for a package to have `amazetux` specific assets for it, right?

Would it make sense to simplify this to basically say that any dot separated identifier is considered non-portable? So `ubuntu` would still work, but `ubuntu.22` would not work. Hmm I guess that's not correct either.
I think we should add the source built native RID into this list then.

Reply:
I don't think we should hard-code these RuntimeIdentifiers here.  They should probably come from or be generated in the Microsoft.NETCoreSdk.BundledVersions.props file which is created in the installer repo.

Reply:
Would putting the list in the (generation for) Microsoft.NETCoreSdk.BundledVersions.props also let source-builds append to it? I have not dealt much with how all the redist/generation comes together.

Reply:
Put up https://github.com/dotnet/installer/pull/16578 for adding `_KnownRuntimeIdentiferPlatforms` items that this will then use.

Reply:
Based on discussion with @dsplaisted, switched to making `ProcessFrameworkReferences` do the processing of known runtime packs and set an output. That should also cover the source-build RID, since it should be part of the runtime pack available RIDs.